flagsmith-engine/features/models.ts

```diff
@@ -96,18 +96,26 @@ export class FeatureStateModel {
         }
         return this.featureSegment.priority < other.featureSegment.priority;
     }
-    
-    getMultivariateValue(identityID: number | string) {
-        const percentageValue = getHashedPercentateForObjIds([
-            this.djangoID || this.featurestateUUID,
-            identityID
-        ]);
 
+    getMultivariateValue(identityID: number | string) {
+        let percentageValue: number | undefined;
         let startPercentage = 0;
-        const sortedF = this.multivariateFeatureStateValues.sort((a, b) =>{
+        const sortedF = this.multivariateFeatureStateValues.sort((a, b) => {
             return a.id - b.id;
         });
         for (const myValue of sortedF) {
+            if (myValue.percentageAllocation === 0) {
+                continue;
+            }
+            if (myValue.percentageAllocation === 100) {
+                return myValue.multivariateFeatureOption.value;
+            }
+            if (percentageValue === undefined) {
+                percentageValue = getHashedPercentateForObjIds([
+                    this.djangoID || this.featurestateUUID,
+                    identityID
+                ]);
+            }
```

Comment:
This looks good, but perhaps the following switch statement would be slightly cleaner - I'm not sure if it's more or less idiomatic for javascript though? 

```suggestion
            switch (myValue.percentageAllocation) {
                case 0:
                    continue;
                case 100:
                    return myValue.multivariateFeatureOption.value;
                default:
                    if (percentageValue === undefined) {
                        percentageValue = getHashedPercentateForObjIds([
                            this.djangoID || this.featurestateUUID,
                            identityID
                        ]);
                    }
            }
```

Reply:
Updated 👍